app/javascript/controllers/submit_proposals_controller.js

```diff
@@ -1,24 +1,74 @@
 import { Controller } from "stimulus"
+import Sortable from "sortablejs"
+import Rails from '@rails/ujs'
 
 export default class extends Controller {
   
-  static targets = [ 'proposalType', 'locationSpecificQuestions', 'locationIds', 'text', 'tabs' ]
+  static targets = [ 'proposalType', 'locationSpecificQuestions', 'locationIds', 'text', 'tabs', 
+                    'dragLocations' ]
   static values = { proposalTypeId: Number, proposal: Number }
   
   connect() {
-    if(this.hasLocationIdsTarget)
+    if(this.hasLocationIdsTarget) {
       this.handleLocationChange(Object.values(this.locationIdsTarget.selectedOptions).map(x => x.value))
+      this.showSelectedLocations()
+    }
   }
 
   handleLocationChange(locations) {
-    if(event && event.type == 'change')
+    if(event && event.type == 'change') {
       locations = [...event.target.selectedOptions].map(opt => opt.value)
+      this.saveLocations()
+    }
+
     fetch(`/proposal_types/${this.proposalTypeIdValue}/location_based_fields?ids=${locations}&proposal_id=${this.proposalValue}`)
       .then(response => response.text())
       .then(html => {
         this.locationSpecificQuestionsTarget.innerHTML = html
       });
   }
+
+  saveLocations() {
+    var _this = this
+    $.post(`/submit_proposals?proposal=${_this.proposalValue}`,
+      $('form#submit_proposal').serialize(), function(data) {
+        _this.showSelectedLocations()
+    })
+  }
+
+  showSelectedLocations() {
+    var _this = this
+    var locationList = ''
+    fetch(`/proposals/${this.proposalValue}/locations.json`)
+    .then(response => response.json())
+    .then(res => {
```

Comment:
Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/birs-math/proposals/pullRequest?prid=7683248)
```suggestion
    .then((res) => {
```